### DISCUSSION TITLE
Wiring of Stepper Driver (H2)

### DISCUSSION THREAD

**Discussion by SteveBon** [kachurovskiy/nanoels#92] [General]
Hi,

I've just built the H2 version but I'm having a problem. The stepper is powered on startup. I can't get it to move at all using the keys/turning the lathe encoder

I have wired the driver thus:
H2----->Driver
STEP –> PUL-
DIR –>   DIR–
ENA ->  ENA-
V5 ->     PUL+, DIR+, ENA+
Is this correct?

I’m using a Stepperonline NEMA23 & DM542T driver

I'm very new to all of this, so sorry if I've made a newbie error

Many thanks
Steve

**Comment by kachurovskiy** [comment]
Hi @SteveBon, can you please post photos of the stepper driver showing all connections and settings as well as it's connection to the NanoEls board? Would also be good to know the voltage you're supplying your stepper driver with. Cheers!

**Comment by SteveBon** [comment]
![IMG_20221203_084642](https://user-images.githubusercontent.com/67836780/205442778-2ecf5efb-149b-431d-9291-5c6fa7b08b39.jpg)
![IMG_20221203_085100](https://user-images.githubusercontent.com/67836780/205442780-bafbb70b-1030-4679-a9cd-26e0bbcdbc2e.jpg)
![IMG_20221203_084847_edit_24923941894633](https://user-images.githubusercontent.com/67836780/205442781-18075711-7220-4468-9953-6ca8dcc27a5f.jpg)
Hopefully these will come out.
I'm driving it using a 24v power supply to the driver. 5v usb to the NanoELS. Steps on the driver are set to 200p.
I was initially using this driver and stepper with a pulse generator to give me a variable speed leadscrew.
When I power up the stepper is energised. I've tried setting the ELS to off but it has no effect.
Many thanks for taking the time to reply 🙂

**Reply by kachurovskiy** [reply]
Awesome, thank you.

If you want stepper to be disabled when resting, set https://github.com/kachurovskiy/nanoels/blob/main/h2/h2.ino#L20 `DISABLE_STEPPER_WHEN_RESTING` to `true`.

It would help to see how the wires are connected on NanoEls, please attach those photos as well if possible.

If you have an oscilloscope, please check if pulses are being sent on STEP wire when you hold the LEFT or RIGHT button pressed on NanoEls. If you don't, a multimeter would also show this (voltage rising from 0 to 0.7 when button pressed).

![pulses](https://user-images.githubusercontent.com/517919/205443992-fcb0058b-b747-4408-9e82-fe0afc1a3bc3.jpg)

**Reply by SteveBon** [reply]
Hello Maxim.
Thank you for the response.
My first image above is of the nanoELS as viewed from the front.
I've now attached a picture from the reverse.
Colours/pins on the NanoELS are as follows:
Red, +5v
Blue, ENA
Green, DIR
Yellow, STEP

Voltage readings with respect to the +5V:
Blue (ENA) 0.05V whatever I press (step, on/off)
Green (DIR) +5V or 0V depending which way I rotate the encoder.
Yellow (STEP) 0V or 0.85V. The 0.85V is when pressing Left or Right.

Many thanks for your time

**Comment by SteveBon** [comment]
![IMG_20221204_180510](https://user-images.githubusercontent.com/67836780/205508086-91521038-10fc-4bf0-b018-8909a7114781.jpg)
Oops, wiring of the Stepper connector from the inside of the box ,

**Comment by kachurovskiy** [comment]
Hi @SteveBon, looks like impulses are properly issued. Can you please check the 5v/24v switch? It should be on 5v position.

![image](https://user-images.githubusercontent.com/517919/205508340-ddb5ef2b-c06e-4d11-a9a2-3c04e38755a3.png)

I would also check the stepper leads - A+, A-, B+, B- - ensure that when leads are disconnected, resistance between A+, A- and B+, B- is low.

Another trick is to disconnect ENA+ and ENA- wires from the driver just in case that it's causing the stepper to be disabled. Cheers!

**Reply by SteveBon** [reply]
OMG, what a newbie mistake to make! I hadn't changed the switch. Thank you so much.
Will fit it to the lathe during the week and report back.

Many thanks again!

**Comment by SteveBon** [comment]
@kachurovskiy Wow Maxim, this thing is fantastic! I've had a little play in my garage tonight. I tried a 1.5mm thread at 550 rpm :-o

Well done & thanks for sharing it.

I want to change the pitch as I'm getting a -ve pitch for right hand threads (personal preference). I think I read somewhere it's the encoder settings that get altered.

**Reply by kachurovskiy** [reply]
Congrats!

> I want to change the pitch as I'm getting a -ve pitch for right hand threads (personal preference). I think I read somewhere it's the encoder settings that get altered.

Sorry, didn't get that. Not sure what's "a -ve pitch for right hand threads".

**Reply by SteveBon** [reply]
I figured out the problem. I bought the encoder from Amazon: https://www.amazon.co.uk/gp/product/B015GYY7XU/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&psc=1

The listing says Green = A phase, White = B phase. Wiring as per this makes left hand threads +ve on the pitch settings. Right hand threads become -ve pitch. This also meant left hand threads went from tailstock to chuck, right hand from chuck to tailstock.

I swapped the green and white wires around. Problem solved.
